### ISSUE TITLE
Panic loading document in v0.5.0

### ISSUE THREAD

**Issue by TristanSpeakEasy** [pb33f/libopenapi#76]
In the below reproducible I am getting 

```
panic: runtime error: index out of range [3] with length 3

goroutine 1 [running]:
github.com/pb33f/libopenapi/index.(*SpecIndex).ExtractRefs(0xc001920900, 0xc00104a8c0, 0xc001043e00, {0xc0031066c0?, 0x7ff0eae23e00?, 0x400?}, 0x7ff0eae9dfff?, 0x40?, {0x0, 0x0})
        /home/trist/go/pkg/mod/github.com/pb33f/libopenapi@v0.5.0/index/spec_index.go:638 +0x2e7b
github.com/pb33f/libopenapi/index.(*SpecIndex).ExtractRefs(0xc001920900, 0xc001043e00, 0xc000f74500, {0xc0031066c0?, 0x7ff0eadb2000?, 0x71c780?}, 0x7ff0eae9dfff?, 0x40?, {0x0, 0x0})
        /home/trist/go/pkg/mod/github.com/pb33f/libopenapi@v0.5.0/index/spec_index.go:634 +0x2af
github.com/pb33f/libopenapi/index.(*SpecIndex).ExtractRefs(0xc001920900, 0xc000f74500, 0xc000f743c0, {0xc000bc6c00?, 0x7ff0d02f8080?, 0x0?}, 0x12c?, 0x40?, {0x0, 0x0})
        /home/trist/go/pkg/mod/github.com/pb33f/libopenapi@v0.5.0/index/spec_index.go:634 +0x2af
github.com/pb33f/libopenapi/index.(*SpecIndex).ExtractRefs(0xc001920900, 0xc000f743c0, 0xc00018c1e0, {0xc001e2bd40?, 0x15?, 0x0?}, 0xc002a397b0?, 0xc8?, {0x0, 0x0})
        /home/trist/go/pkg/mod/github.com/pb33f/libopenapi@v0.5.0/index/spec_index.go:634 +0x2af
github.com/pb33f/libopenapi/index.(*SpecIndex).ExtractRefs(0xc001920900, 0xc00018c1e0, 0xc00018c000, {0xc002a39998?, 0x890000000040eb47?, 0x2?}, 0xc0004c04e0?, 0x80?, {0x0, 0x0})
        /home/trist/go/pkg/mod/github.com/pb33f/libopenapi@v0.5.0/index/spec_index.go:634 +0x2af
github.com/pb33f/libopenapi/index.NewSpecIndex(0xc00018c000)
        /home/trist/go/pkg/mod/github.com/pb33f/libopenapi@v0.5.0/index/spec_index.go:254 +0x773
github.com/pb33f/libopenapi/datamodel/low/v3.CreateDocument(0xc00018c0a0)
        /home/trist/go/pkg/mod/github.com/pb33f/libopenapi@v0.5.0/datamodel/low/v3/create_document.go:25 +0x17d
github.com/pb33f/libopenapi.(*document).BuildV3Model(0xc000756000?)
        /home/trist/go/pkg/mod/github.com/pb33f/libopenapi@v0.5.0/document.go:153 +0x4e
main.main()
        /home/trist/workspace/scratch/main.go:32 +0x37
exit status 2
```

```go
package main

import (
	"fmt"
	"io"
	"net/http"

	"github.com/pb33f/libopenapi"
)

func read() []byte {
	res, err := http.Get("https://raw.githubusercontent.com/speakeasy-api/openapi-directory/main/APIs/atlassian.com/jira/1001.0.0-SNAPSHOT/openapi.yaml")
	if err != nil {
		panic(err)
	}
	defer res.Body.Close()

	data, err := io.ReadAll(res.Body)
	if err != nil {
		panic(err)
	}

	return data
}

func main() {
	doc, err := libopenapi.NewDocument(read())
	if err != nil {
		panic(err)
	}

	m, errs := doc.BuildV3Model()
	if errs != nil {
		panic(errs)
	}

	fmt.Println(m)
}
```

**Comment by TristanSpeakEasy** [comment]
https://raw.githubusercontent.com/speakeasy-api/openapi-directory/main/APIs/bbc.com/1.0.0/openapi.yaml also causes a similar panic

**Comment by TristanSpeakEasy** [comment]
another example https://raw.githubusercontent.com/speakeasy-api/openapi-directory/main/APIs/microsoft.com/graph/beta/openapi.yaml

**Comment by daveshanley** [comment]
Oh dear. I'll have a look first thing in the morning when I come online.

**Comment by daveshanley** [comment]
This should be fixed in `v0.5.1`

**Comment by TristanSpeakEasy** [comment]
Can confirm this is fixed. Thanks Dave
